nilearn/signal.py

```diff
@@ -473,7 +473,16 @@ def clean(signals, sessions=None, detrend=True, standardize=True,
     # Remove confounds
     if confounds is not None:
         confounds = _ensure_float(confounds)
-        confounds = _standardize(confounds, normalize=True, detrend=detrend)
+        confounds = _standardize(confounds, normalize=standardize,
+                                 detrend=detrend)
+        if not standardize:
+            # Improve numerical stability by controlling the range of
+            # confounds. We don't rely on _standardize as it removes any
+            # constant contribution to confounds
+            confound_max = np.maximum(confounds.max(axis=0),
+                                      -confounds.min(axis=0))
```

Comment:
that makes a temporary array, but probably no problem since it wouldn't be
big

On Tue, Jan 26, 2016 at 10:40 AM, Alexandre Abraham <
notifications@github.com> wrote:

> In nilearn/signal.py
> <https://github.com/nilearn/nilearn/pull/963#discussion_r50812317>:
>
> > @@ -473,7 +473,16 @@ def clean(signals, sessions=None, detrend=True, standardize=True,
> >      # Remove confounds
> >      if confounds is not None:
> >          confounds = _ensure_float(confounds)
> > -        confounds = _standardize(confounds, normalize=True, detrend=detrend)
> > +        confounds = _standardize(confounds, normalize=standardize,
> > +                                 detrend=detrend)
> > +        if not standardize:
> > +            # Improve numerical stability by controlling the range of
> > +            # confounds. We don't rely on _standardize as it removes any
> > +            # constant contribution to confounds
> > +            confound_max = np.maximum(confounds.max(axis=0),
> > +                                      -confounds.min(axis=0))
>
> Personally, I prefer np.abs(confounds).max(axis=0).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nilearn/nilearn/pull/963/files#r50812317>.
>